### PR TITLE
Fixes #13011 - use standard way to join with task

### PR DESCRIPTION
**Pull request by iNecas**
Otherwise, the scoped search has issues producing valid sql

### CODE REVIEW

File: app/models/job_invocation.rb [hunk 1]
```diff
@@ -36,7 +36,7 @@ class JobInvocation < ActiveRecord::Base
   belongs_to :triggering, :class_name => 'ForemanTasks::Triggering'
   has_one :recurring_logic, :through => :triggering, :class_name => 'ForemanTasks::RecurringLogic'
 
-  scope :with_task, -> { joins('LEFT JOIN foreman_tasks_tasks ON foreman_tasks_tasks.id = job_invocations.task_id') }
+  scope :with_task, -> { joins(:task) }
```
Comment (ohadlevy) [inline app/models/job_invocation.rb]: :+1: what was the original reasoning before?

Comment (ares) [inline app/models/job_invocation.rb]: it was originally added in https://github.com/theforeman/foreman_remote_execution/commit/a9c05c8c7dfe8d6dfcf91c782d9b9985f80a19fb#diff-d3ff599b762e4ff95c9b2776d2be1079R32 in https://github.com/theforeman/foreman_remote_execution/pull/76, maybe @adamruzicka would remember... could be some issue in rails 3 AR or it just slipped through

Comment (iNecas) [inline app/models/job_invocation.rb]: Ok, we needed the outer join, which turned out to be present in Rails 4 via `references`

**Comment by iNecas** [comment]
In case we are dropping Foreman 1.10 starting iteration 5, we can remove the 1.10 compatibility condition

**Comment by ares** [comment]
ACK

**Pull request merged**